src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -143,6 +146,21 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
   } = ReactFiberScheduler(config);
 
   function scheduleTopLevelUpdate(current : Fiber, element : ReactNodeList, callback : ?Function) {
+    if (__DEV__) {
+      const owner = ReactCurrentOwner.current;
+      if (owner && typeof owner.tag === 'number') {
```

Comment:
This extra tag check here is an indication that you're over using the ReactCurrentOwner concept. Should be fixed if you just do a render specific dev only flag for this specific use case.